docs/INSTALL.md

```diff
@@ -3,35 +3,226 @@ SPDX-License-Identifier: Apache-2.0
+    diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+    index 086ad2f..79d81bd 100755
+    --- a/scripts/docker_setup.sh
+    +++ b/scripts/docker_setup.sh
+    @@ -16,7 +16,7 @@ bessd_port=10514
+    mode="dpdk"
+    #mode="af_xdp"
+    #mode="af_packet"
+    -#mode="sim"
+    +mode="sim"
+
+    # Gateway interface(s)
+    #
+    ```
+
+3. Start installation from UPF's root directory
+
+    ```bash
+    ./scripts/docker_setup.sh
+    ```
+
+4. Insert rules into PDR and FAR tables
+
+    Use gRPC sim mode to directly install PFCP forwarding rules via gRPC API (works only for BESS)
+
+    ```bash
+    docker exec -ti bess-pfcpiface pfcpiface -config /conf/upf.json -simulate create
+    ```
+
+    OR
+
+    Use the [pfcpsim](https://github.com/omec-project/pfcpsim) tool to generate PFCP messages towards the PFCP Agent.
+
+5. (optional) Observe the pipeline in GUI
+
+   To display GUI of the pipeline visit [http://[hostip]:8000](http://[hostip]:8000)
+
+
+## Installation DPDK mode
+
+To install the UPF in DPDK mode, the following changes are required:
+
+1. Make appropriate changes in configuration file
+
+    ```patch
+    $ git diff conf/upf.json
+    diff --git a/conf/upf.json b/conf/upf.json
+    index c6531cc..335e33b 100644
+    --- a/conf/upf.json
+    +++ b/conf/upf.json
+    @@ -51,7 +51,7 @@
+         "gtppsc": false,
+
+         "": "Enable Intel Dynamic Device Personalization (DDP)",
+    -    "ddp": false,
+    +    "ddp": true,
+
+         "": "Telemetrics-See this link for details: https://github.com/omec-project/bess/blob/master/bessctl/module_tests/timestamp.py",
+         "measure_upf": true,
+    @@ -61,12 +61,12 @@
+
+         "": "Gateway interfaces",
+         "access": {
+    -        "ifname": "ens803f2"
+    +        "ifname": "access"
+         },
+
+         "": "UE IP Natting. Update the line below to `\"ip_masquerade\": \"<ip> [or <ip>]\"` to enable",
+         "core": {
+    -        "ifname": "ens803f3",
+    +        "ifname": "core",
+             "": "ip_masquerade: 18.0.0.1 or 18.0.0.2 or 18.0.0.3"
+         },
+    ```
+
+2. Update parameters in script file
+> NOTE: Please remeber to follow [these](dpdk-configuration.md) directions before executing the next step
+
+    ```patch
+    $ git diff scripts/docker_setup.sh
+            diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+    index f5946c1..637d62b 100755
+    --- a/scripts/docker_setup.sh
+    +++ b/scripts/docker_setup.sh
+    @@ -22,7 +22,7 @@ mode="dpdk"
+     # Gateway interface(s)
+     #
+     # In the order of ("s1u" "sgi")
+    -ifaces=("ens803f2" "ens803f3")
+    +ifaces=("access" "core")
+
+     # Static IP addresses of gateway interface(s) in cidr format
+     #
+    @@ -32,7 +32,7 @@ ipaddrs=(198.18.0.1/30 198.19.0.1/30)
+     # MAC addresses of gateway interface(s)
+     #
+     # In the order of (s1u sgi)
+    -macaddrs=(9e:b2:d3:34:ab:27 c2:9c:55:d4:8a:f6)
+    +macaddrs=(b4:96:91:b1:ff:f0 b4:96:91:b1:ff:f1)
```

Comment:
got bit confused after proposed changes in macaddrs/others , but when i saw dpdk config, i could link it up , but it wont be easy for newcomers/first time setters to understand what are the specific parameters required from dpdk, appreciate if we can specifically mention - i.e macaddrs, dev etc need to be derived by runnin dpdk tools/others etc.

Reply:
This is why I added the `note` right after step 2. I am going to add a bit more text there to be more explicit
`> NOTE: Please remember to follow [these](dpdk-configuration.md) directions before executing the next step`